ghcide/src/Development/IDE/Core/Compile.hs

```diff
@@ -942,8 +942,9 @@ indexHieFile se mod_summary srcPath !hash hf = do
             Nothing -> pure Nothing
             Just env -> LSP.runLspT env $ do
               u <- LSP.ProgressToken . LSP.InR . T.pack . show . hashUnique <$> liftIO Unique.newUnique
-              -- TODO: Wait for the progress create response to use the token
-              _ <- LSP.sendRequest LSP.SMethod_WindowWorkDoneProgressCreate (LSP.WorkDoneProgressCreateParams u) (const $ pure ())
+              b <- liftIO newBarrier
+              void $ LSP.sendRequest LSP.SMethod_WindowWorkDoneProgressCreate (LSP.WorkDoneProgressCreateParams u) $ liftIO . signalBarrier b
+              ready <- liftIO $ waitBarrier b
```

Comment:
I don't know if its a good idea to delay indexing by an unbounded amount of time waiting for the client to respond with a progress token. Perhaps the progress reporting should be on another thread which doesn't block indexing.

Reply:
yes, but indexHieFile return without reportProgress actually report done🤔.